test/unit/picolibrary/indicator/gpio_output_pin_fixed_intensity_indicator/main.cc

```diff
@@ -0,0 +1,176 @@
+
+    EXPECT_CALL( indicator, initialize( _ ) ).WillOnce( Return( error ) );
+
+    auto const result = indicator.initialize( random<Initial_Indicator_State>() );
+
+    EXPECT_TRUE( result.is_error() );
+    EXPECT_EQ( result.error(), error );
+}
+
+/**
+ * \brief Verify
+ *        picolibrary::Indicator::GPIO_Output_Pin_Fixed_Intensity_Indicator::initialize()
+ *        works properly.
+ */
+TEST( initialize, worksProperly )
+{
+    struct {
+        Initial_Indicator_State requested_state;
+        Initial_Pin_State       actual_state;
+    } const test_cases[]{
+        { Initial_Indicator_State::ILLUMINATED, Initial_Pin_State::HIGH },
+        { Initial_Indicator_State::EXTINGUISHED, Initial_Pin_State::LOW },
+    };
+
+    for ( auto const test_case : test_cases ) {
+        auto indicator = Indicator{};
+
+        EXPECT_CALL( indicator, initialize( test_case.actual_state ) )
+            .WillOnce( Return( Result<Void, Error_Code>{} ) );
+
+        EXPECT_FALSE( indicator.initialize( test_case.requested_state ).is_error() );
+    } // for
+}
+
+/**
+ * \brief Verify
+ *        picolibrary::Indicator::GPIO_Output_Pin_Fixed_Intensity_Indicator::illuminate()
+ *        properly handles a state transition error.
+ */
+TEST( illuminate, stateTransitionError )
+{
+    auto indicator = Indicator{};
+
+    auto const error = random<Mock_Error>();
+
+    EXPECT_CALL( indicator, transition_to_high() ).WillOnce( Return( error ) );
+
+    auto const result = indicator.illuminate();
+
+    EXPECT_TRUE( result.is_error() );
+    EXPECT_EQ( result.error(), error );
+}
+
+/**
+ * \brief Verify
+ *        picolibrary::Indicator::GPIO_Output_Pin_Fixed_Intensity_Indicator::illuminate()
+ *        works properly.
+ */
+TEST( illuminate, worksProperly )
+{
+    auto indicator = Indicator{};
+
+    EXPECT_CALL( indicator, transition_to_high() ).WillOnce( Return( Result<Void, Error_Code>{} ) );
+
+    EXPECT_FALSE( indicator.illuminate().is_error() );
+}
+
+/**
+ * \brief Verify
+ *        picolibrary::Indicator::GPIO_Output_Pin_Fixed_Intensity_Indicator::extinguish()
+ *        properly handles a state transition error.
+ */
+TEST( extinguish, stateTransitionError )
+{
+    auto indicator = Indicator{};
+
+    auto const error = random<Mock_Error>();
+
+    EXPECT_CALL( indicator, transition_to_low() ).WillOnce( Return( error ) );
+
+    auto const result = indicator.extinguish();
+
+    EXPECT_TRUE( result.is_error() );
+    EXPECT_EQ( result.error(), error );
+}
+
+/**
+ * \brief Verify
+ *        picolibrary::Indicator::GPIO_Output_Pin_Fixed_Intensity_Indicator::extinguish()
+ *        works properly.
+ */
+TEST( extinguish, worksProperly )
+{
+    auto indicator = Indicator{};
+
+    EXPECT_CALL( indicator, transition_to_low() ).WillOnce( Return( Result<Void, Error_Code>{} ) );
+
+    EXPECT_FALSE( indicator.extinguish().is_error() );
+}
+
+/**
+ * \brief Execute the picolibrary::Indicator::GPIO_Output_Pin_Fixed_Intensity_Indicator unit tests.
```

Comment:
Is the text width of this line 90 or less? Please reformat the line if it is not.